source/extensions/common/aws/signer_base_impl.cc

```diff
@@ -124,13 +125,24 @@ void SignerBaseImpl::sign(Http::RequestHeaderMap& headers, const std::string& co
     // Append signature to existing query string
     query_params.add(SignatureQueryParameters::get().AmzSignature, signature);
     headers.setPath(query_params.replaceQueryString(headers.Path()->value()));
-    ENVOY_LOG(debug, "Query string appended to path");
+    // Sanitize logged query string
+    query_params.overwrite(SignatureQueryParameters::get().AmzSignature, "*****");
+    query_params.overwrite(SignatureQueryParameters::get().AmzSecurityToken, "*****");
```

Comment:
Would it handle the condition if  `AmzSecurityToken` is not set on query_params?

Reply:
good catch ty